src/main/java/com/cqse/teamscaleupload/ReportPatternManager.java

```diff
@@ -13,23 +14,49 @@
 import java.util.stream.Collectors;
 
 /**
- * This class manages the given file formats and
- * corresponding report file patterns.
+ * This class provides the functionality to resolve the report file patterns and formats
+ * from the command line to a map from report formats to actual files which can then be
+ * used for the upload to Teamscale.
```

Comment:
You changed this class so that it now only contains static methods. I'd thus prefer we rename this `ReportPatternUtils` as it no longer "manages" anything ;-)

Reply:
Alternative: make the methods non-static and re-introduce the field you had previously for the List-Map. Then you don't have to pass that to every private utility method. It's up to you.